packages/grid/x-grid-modules/src/models/featureMode.ts

```diff
@@ -0,0 +1,4 @@
+export enum FeatureMode {
+  Client = 'client',
+  Server = 'server',
+}
```

Comment:
Do we even need a public enum for this? Do we even need a private enum for this?

```suggestion
export enum FeatureMode {
  client = 'client',
  server = 'server',
}
```

Reply:
it's cleaner and it avoids using magic strings. I need it in several places

Reply:
same can't commit this change. Needs to be done across the repo

Reply:
Ok, so you advocate for using a variable for internal usage. I used to do that often when using JavaScript only. I always assumed that with TypeScript, it would only duplicate with the types.

Regarding the public API? should we expose it?

Reply:
Well we can expose it and let users decide if they want to use it or not. 
I don't see why we should add a prefix like Grid or GridPagination.
I'm planning to use this enum in sorting too and probably filtering.

Users won't have name clashing, as they should use the lib in isolation of other components. It's not like they are going to use all their components in one file. Libraries don't prefix all their types with the library name.

Reply:
What about we keep it private? We should be cautious with providing two ways to do the same things, it creates confusion around what's the correct way. It wouldn't be consistent with how we handle the API with the other components. We don't expose variables for each enum the components supports. Does it solve a problem for users?

Regarding prefixing, I think that it's important to consider that each exposed variable is in a global namespace, containing all the components. So far, we have enforced no collison on the exposed types. I think that we should continue to do so. It's important for 1. people that grep in their codebase for refactoring 2. re-exporting the types from `@material-ui/x`.

Reply:
It's also important for the documentation, to make it searchable. It's basically something we started discussing in https://github.com/mui-org/material-ui-pickers/issues/1778

Reply:
Well see it a bit like passing a constant instead of a string directly. In the end, it's a string.

Ok for prefixing with 'Grid' if you want. I don't mind. 

I thought they would do something a bit like this, so it works with tree shaking

`import {FeatureMode} from '@material-ui/x/grid'`

Reply:
Yeah, I think that Grid would be enough :), mind that we had prefixes about `ExpensionPanel`, much longer. 

Would you do? 

```jsx
import { Button, ButtonVariant } from '@material-ui/core';

<Button variant={ButtonVariant.outlined}>
```

Reply:
nope I would do something like 
`<Button variant={theme.variant}` />

In the button code, I wouldn't use "variant" as my code would be full of if(something === "variant")
As if "variant" changes then I have to rename it everywhere. Hence constants  or enum.